srcCpp/Infrastructure_common.cxx

```diff
@@ -14,6 +14,20 @@ HANDLE PerftestTimer::_hTimer = NULL;
 #endif
 void (*PerftestTimer::_handlerFunction)(void) = NULL;
 
+void *PerftestTimer::waitAndExecuteHandler(void *timer) 
+{
+    unsigned int *t = static_cast<unsigned int *>(timer);
+
+    // Sleep for t milliseconds
+    PerftestClock::milliSleep(*t);
+
+    // Call the scheduled function with the args
+    PerftestTimer::timeoutTask(0);
+
+    // Free up space from the allocation of the uint
+    delete t;
```

Comment:
> It's a good practice check if t is null before calling delete.

Good point, fixing it.

> Even so, I don't like give to a function the responsibility to remove something that the function itself has not created. (With the exception of a destructor)

I agree. The problem is that `PerftestTimer::setTimeout` receives `unsigned int executionTimeInSeconds` as a parameter of the function. This is a copy of the original value that will live in the scope of the `setTimeout` function. Then `PerftestThread_new` function receives `RTIOsapiThreadOnSpawnedMethod` as 4th parameter which is a typedef for `void *(*RTIOsapiThreadOnSpawnedMethod)(void *threadParam)`. Since it receives a pointer to void as a parameter, we need to pass a pointer to that uint value for `executionTimeInSeconds`: that's why `new uint(executionTimeInSeconds * 1000u)` is used.

Because `PerftestTimer::setTimeout` can finish earlier than the thread executes the sleep method (`PerftestClock::milliSleep`), we cannot guarantee that the object is still there to be referenced if we delete the object in `PerftestTimer::setTimeout()` right after creating the thread.

Reply:
Good explanation!. Anyway, you can give the variable `executionTimeInSeconds * 1000u` to the class and destroy it on the destructor. Because this is a singleton the destructor will be called at the end of the program.

Reply:
I've taken a different approach. Now the function receives an struct by reference containing both the timer and the function to execute afterwards.

Thanks!